### PR TITLE
MODPUBSUB-215: Replace Future<Boolean> by Future<Void> in ConsumerService

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[MODPUBSUB-215](https://issues.folio.org/browse/MODPUBSUB-215)
_Replace Future<Boolean> by Future<Void> in ConsumerService_

### CODE REVIEW

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
    *
    * @param eventTypes list of event types that specified module is subscribing to receive
    * @param params     Okapi connection params
-   * @return future with true if succeeded
+   * @return succeeded Future
```
Comment (sherzodn) [inline mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java:18]:
```suggestion
   * @return a future completed with a result
```

File: mod-pubsub-server/src/main/java/org/folio/services/MessagingModuleService.java [hunk 2]
```diff
@@ -48,9 +48,9 @@
    *
    * @param subscriberDescriptor subscriber descriptor
    * @param params               Okapi connection params
-   * @return future with boolean
+   * @return succeeded Future
```
Comment (sherzodn) [inline mod-pubsub-server/src/main/java/org/folio/services/MessagingModuleService.java:51]:
```suggestion
   * @return a future completed with a result
```

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java (hunk 1 above)
Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java]: edited

File: mod-pubsub-server/src/main/java/org/folio/services/MessagingModuleService.java (hunk 2 above)
Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/MessagingModuleService.java]: edited

**Review by Mykyta-Varenyk — APPROVED** [review]

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 3]
```diff
@@ -69,40 +71,32 @@ public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
...
-    GenericCompositeFuture.all(list)
-      .onComplete(ar -> {
-        if (ar.succeeded()) {
-          result.complete(true);
-        } else {
-          result.fail(ar.cause());
-        }
-      });
-    return result.future();
+        return promise.future();
```
Comment (julianladisch) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java:97]: The promise is no longer needed:

```
       .map(topic -> {
         consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, topic);
         return KafkaConsumer.<String, String>create(vertx, consumerProps)
           .handler(getEventReceivedHandler(params))
           .subscribe(topic)
           .onSuccess(x -> {
             cache.addSubscription(topic);
             LOGGER.info(format("Subscribed to topic {%s}", topic));
           })
           .onFailure(e -> {
             LOGGER.error(format("Could not subscribe to some of the topic {%s}", topic), e);
           });
```

Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: fixed, thanks

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request closed without merging**